### PR TITLE
Added option for auto-injector to chemmaster

### PR DESCRIPTION
**Pull request by Aivaria**
[input wanted]

## About the PR 
It adds the option of boxed and single auto-injectors to the chemmaster. Includes the Hypospray whitelist (emagging chemmaster makes it ignore the whitelist). 
Adds a big container without limits, that doesnt fit in the belt slot (weight 3), and a small one with a capacity of 3, which can be added to belt

## Why's this needed? 
i always thought its an missing feature, as these could easily be handed out to people.

## Changelog

```changelog
(u)Aivaria + littleBugHunter
(*)Added the options for single and multiple auto-injectors to chemmaster. Chemicals are restricted by whitelist, but can be emagged to allow any chem.
```

### CODE REVIEW

**Comment by DisturbHerb** [comment]
maybe change the changelog to something more like this

```
(u)Aivaria + littleBugHunter
(*)Added the options for single and multiple auto-injectos to chemmaster
```

File: code/obj/item/generic_box.dm [hunk 1]
```diff
@@ -304,6 +304,48 @@
 			item_amount = 10
 			max_item_amount = 10
 
+	autoinjector
+		name = "box of emergency auto-injectors"
+		name = "box of patches"
```
Comment (TobleroneSwordfish) [inline code/obj/item/generic_box.dm:309]:
```suggestion
```

File: code/modules/chemistry/Chemistry-Machinery.dm [hunk 2]
```diff
@@ -631,6 +731,8 @@
 				dat += "<A href='?src=\ref[src];createcan=1'>Create can (50 units max)</A><BR>"
 				dat += "<A href='?src=\ref[src];createpatch=1'>Create patch (30 units max)</A><BR>"
 				dat += "<A href='?src=\ref[src];multipatch=1'>Create multiple patches (5 units min)</A> Box: <A href='?src=\ref[src];togglepatchbox=1'>[src.patch_box ? "Yes" : "No"]</A><BR>"
+				dat += "<A href='?src=\ref[src];createautoinjector=1'>Create emergeny auto-injector (10 units max)</A><BR>"
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Machinery.dm:734]:
```suggestion
				dat += "<A href='?src=\ref[src];createautoinjector=1'>Create emergency auto-injector (10 units max)</A><BR>"
```

File: code/modules/chemistry/Chemistry-Machinery.dm [hunk 3]
```diff
@@ -631,6 +731,8 @@
...
+				dat += "<A href='?src=\ref[src];multiautoinjector=1'>Create multiple emergeny auto-injectors (5 units min)</A>  Box: <A href='?src=\ref[src];toggleinjectorbox=1'>[src.injector_box==1 ? "Yes (Big)" : (src.injector_box==2 ? "Yes (Small)" : "No")]</A><BR>"
```
Comment (TobleroneSwordfish) [inline code/modules/chemistry/Chemistry-Machinery.dm:735]:
```suggestion
				dat += "<A href='?src=\ref[src];multiautoinjector=1'>Create multiple emergency auto-injectors (5 units min)</A>  Box: <A href='?src=\ref[src];toggleinjectorbox=1'>[src.injector_box==1 ? "Yes (Big)" : (src.injector_box==2 ? "Yes (Small)" : "No")]</A><BR>"
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
tgui chemmaster when

**Comment by Wisemonster** [comment]
Might wanna add the emag function to the changelog.

**Comment by flrsh** [comment]
I don't think the chemmaster should be able to create autoinjectors. It doesn't really make sense to me and I like autoinjectors only being found in portananomeds given their convenience. I am also not a fan of emagging letting you use any chems - autoinjectors are already the only chemical vehicle you can really trust without a spectroscopic scanner.

**Comment by Aivaria** [comment]
> I don't think the chemmaster should be able to create autoinjectors. It doesn't really make sense to me and I like autoinjectors only being found in portananomeds given their convenience. I am also not a fan of emagging letting you use any chems - autoinjectors are already the only chemical vehicle you can really trust without a spectroscopic scanner.

currently they would behave equivalent to one use, low volume hyposprays. 
The intention (from medic and rp play view) is the option, to add an alternative to pills you can savely hand out to your fellow miners or security, they can use in case of emergency. (like potassium iodide to miners, in case they get radiated). they are quite low volume, and one use, so you wouldnt hand out a bottle of like 10 pills, and never see that person again, instead you hand out like 1 or a box of 3, so the patient might come back, and so get some interaction again.
Also, since it uses the hypospray whitelist, the chemicals usualy shouldnt be (too) dangerous in them (there are even more dangerous one available in porta-med.

But honestly, i dont know in which way it might affect gameplay in the end. i would imagine, that botany and medbay on RP servers would profit from this (already had some feedback, that love this idea), but i dont know how this would end in the hands of an baddie or how it behaves on classes, as i lack the experience in those field.

**Comment by Aivaria** [comment]
Maybe, to add an different argument - currently you could also use patches kinda the same way, like you could use the auto-injector in this PR. with the difference, these give an one shot injection, of up to 10u, while patches add a content of up to 30u each to the person over time. To get this more fun, you can use patches 'out lof the box' (literaly). If the chemmaster is emagged, it can create any chem patch, without them having a delay while attaching them to a person.

i would say, lets test how this ends out.
if its too over powered or abused, we may shrink things down, like reduce that size 3 box, to size 2 ( it was also the first size i had in my mind when working with them ), and increase the weight class of the 'unlimited' box, so it doesnt fit into the backpack anymore (and maybe make it so, it contains the smaller size boxes, so you cant run arround with an giant box, and pump chem after chem into people from the large box).
If thats still well to overused / abused, then my offer is: ping me in discord, and i provide the patch to remove this feature asap i can.

**Comment by TDHooligan** [comment]
probly worth a discussion thread. i've thought of the idea while adding the cans to the ChemMaster. I think they intrude on the territory of the hypospray a little much, and if you add small-sized boxes of 3 you functionally have something better than a hypospray until you need to refill them. 
also as flourish says - they currently are the only reliable source of medicines.

**Comment by UrsulaMejor** [comment]
Don't want the ability to emag them to bypass whitelist. Use a hypospray for that; autoinjectors should be reliable "safe" things that do not need to be questioned.

**Comment by ZeWaka** [comment]
not a fan re: emily, flourish

**Comment by flrsh** [comment]
Devs don't seem to be a fan of this, closing.

**Pull request closed without merging**